src/types.ts

```diff
@@ -0,0 +1,62 @@
+export type Auth0APIClient = {
```

Comment:
I mentioned this in the PR description, but I'd like to call out again that I usually don't like to separate out types in this way. My preference is to keep the types as close to their functional code counterpart as possible. However, I discovered that these shared types are used widely across the codebase and didn't "belong" in any one location. And to avoid reorganizing the directories, I just opted for this general file. Over the course of this migration I'll be on the lookout for a better spot for these types, but I suspect this file is here for the long haul.